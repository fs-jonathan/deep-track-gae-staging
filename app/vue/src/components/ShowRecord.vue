<template>
  <div class="showRecord">
    <div id="loader">
      <vue-element-loading :active="loading" spinner="bar-fade-scale" :is-full-screen="true"/>
    </div>

    <div class="bg-blue-lightest border-t border-b border-blue text-blue-dark px-4 py-3" v-if="error">
      <p class="text-sm">{{ error }}</p>
    </div>

    <div style="display:none" id="content" v-if="results">
      <div v-for="(result, key, index) in results" :key="index">
        <DetailRow :result="result" />
      </div>
    </div>
  </div>
</template>

<script>
import DetailRow from './DetailRow'

export default {
  name: 'ShowRecord',
  data: () => ({
    loading: false,
    results: null,
    error: null
  }),
  components: {
    DetailRow,
  },
  created () {
    // fetch the data when the view is created and the data is already being observed
    this.getJson(this.$route.params.index)
  },
  methods: {
    getJson: function(id) {
      this.loading = true

      this.$axios.post('/getDetail', { "message": id })
           .then(response => {
             this.results = response.data;
           })
           .catch(err => {
             this.error = err.toString();
           })
           .finally(() => {
             this.loading = false
             document.getElementById("content").style.display = "block";
           })
    }
  }
}
</script>

<style scoped>
</style>
