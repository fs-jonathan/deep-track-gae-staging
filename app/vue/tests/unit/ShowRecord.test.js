import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import ShowRecord from '@/components/ShowRecord.vue'
import VueElementLoading from 'vue-element-loading'
import flushPromises from 'flush-promises'

const localVue = createLocalVue()
localVue.use(VueRouter)
localVue.component('VueElementLoading', VueElementLoading)

const router = new VueRouter()

const data = [{"id":6,"title":"見積もり収益額","subtitle":"","cost":9,"compare":0,"rate":0},{"id":7,"title":"ページビュー","subtitle":"","cost":193,"compare":0,"rate":0},{"id":8,"title":"表示回数","subtitle":"","cost":8,"compare":0,"rate":0},{"id":9,"title":"ページCTR","subtitle":"","cost":0.7269779556872005,"compare":0,"rate":0},{"id":10,"title":"クリック率","subtitle":"","cost":0.8896918783667795,"compare":0,"rate":0},{"id":11,"title":"カバレッジ","subtitle":"","cost":0.5157602624061407,"compare":0,"rate":0}]

describe('ShowRecord.vue', () => {
  it('renders correctly', async () => {
    const wrapper = shallowMount(ShowRecord, {
      localVue,
      router,
      attachToDocument: true,
      mocks: {
        $axios: {
          post: () => Promise.resolve({ data: data })
        }
      }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
    await flushPromises()
    expect(wrapper.element).toMatchSnapshot()
  })
})
