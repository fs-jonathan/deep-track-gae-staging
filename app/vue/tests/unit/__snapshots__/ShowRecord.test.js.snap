// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowRecord.vue renders correctly 1`] = `
<div
  class="showRecord"
>
  <div
    id="loader"
  >
    <vueelementloading-stub
      isfullscreen="true"
      spinner="bar-fade-scale"
    />
  </div>
   
  <!---->
   
  <div
    id="content"
    style="display: block;"
  >
    <div>
      <detailrow-stub
        result="[object Object]"
      />
    </div>
    <div>
      <detailrow-stub
        result="[object Object]"
      />
    </div>
    <div>
      <detailrow-stub
        result="[object Object]"
      />
    </div>
    <div>
      <detailrow-stub
        result="[object Object]"
      />
    </div>
    <div>
      <detailrow-stub
        result="[object Object]"
      />
    </div>
    <div>
      <detailrow-stub
        result="[object Object]"
      />
    </div>
  </div>
</div>
`;
